setupJest.js

```diff
@@ -1 +1,48 @@
+import iface from 'jsdom/lib/jsdom/living/generated/Element'
+
+Element.prototype.insertAdjacentElement = function insertAdjacentElement(position, node) {
+    position = position.toLowerCase();
+
+    let context;
+    switch (position) {
+        case "beforebegin":
+        case "afterend": {
+            context = this.parentNode;
+            if (context === null || context.nodeType === NODE_TYPE.DOCUMENT_NODE) {
+              throw new DOMException(DOMException.NO_MODIFICATION_ALLOWED_ERR, "Cannot insert HTML adjacent to " +
+                "parent-less nodes or children of document nodes.");
+            }
+            break;
+        }
+        case "afterbegin":
+        case "beforeend": {
+            context = this;
+            break;
+        }
+        default: {
+            throw new DOMException(DOMException.SYNTAX_ERR, "Must provide one of \"beforebegin\", \"afterend\", " +"\"afterbegin\", or \"beforeend\".");
+        }
+    }
```

Comment:
`context` seems unused in the code below, so the stuff above could be removed or (better) made simpler (it's just testing for invalid scenarios).